### PR TITLE
feat(product): add a testing module for easily mocking the facades

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Any time a facade is used, any fields used from the facade must be mocked in each test.

## What is the new behavior?
A testing module is provided that mocks out all fields of the facade for you with `BehaviorSubject`s

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/product/src/facades/product/product-facade.interface.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Observable } from 'rxjs';
+
+import { DaffProduct } from '../../models/product';
+
+export interface DaffProductFacadeInterface<T extends DaffProduct = DaffProduct> {
+	loading$: Observable<boolean>;
+	product$: Observable<T>;
+	dispatch: Function;
```
Comment (damienwebdev) [inline libs/product/src/facades/product/product-facade.interface.ts:8]: Can this have a more explicit type than `Function`?

File: libs/product/src/facades/product-grid/product-grid-facade.interface.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Observable } from 'rxjs';
+
+import { DaffProduct } from '../../models/product';
+
+export interface DaffProductGridFacadeInterface<T extends DaffProduct = DaffProduct> {
+	loading$: Observable<boolean>;
+	products$: Observable<T[]>;
+	dispatch: Function;
```
Comment (damienwebdev) [inline libs/product/src/facades/product-grid/product-grid-facade.interface.ts:8]: Can this have a more explicit type than `Function`?

File: libs/product/src/facades/product/product-facade.interface.ts (hunk 1 above)
Comment (lderrickable) [reply libs/product/src/facades/product/product-facade.interface.ts]: done

File: libs/product/src/facades/product-grid/product-grid-facade.interface.ts (hunk 2 above)
Comment (lderrickable) [reply libs/product/src/facades/product-grid/product-grid-facade.interface.ts]: done

**Pull request merged**